### ISSUE TITLE
Presence of an argument?

### ISSUE THREAD

**Issue by charlesspitzer** [yargs/yargs#1334]
Say I have the arguments --abc N. If I make the default for abc be 5, then yargs will create the .abc value with 5. However, this doesn't tell me if --abc was actually seen on the command line. Is there a way to see an argument was given at all?

**Comment by aorinevo** [comment]
If I recall correctly, an options flag with no value is assumed to be Boolean.  So it should show up on the args object.

That is args.abc === true if abc option is added to the command.

**Comment by charlesspitzer** [comment]
that's not exactly what i'm after.

if i have 
const yargs = require('yargs')
   .option('abc', {default: 5,type: 'number'})
   .option('def', {default:false,type:'boolean'})
   .option('ghi', {default:false,type:'boolean'});;

and the command line
cmd --ghi

there doesn't seem to be a way to tell if --abc was specified, nor if --def is using the default value, because yargs.abc will === 5 and yargs.def will be false in this cmd invocation.

what's needed is some boolean on each argument to tell if it was actually present on the command line or if the default created the field in the yargs object.

**Comment by bcoe** [comment]
@charlesspitzer out of curiosity, what's your use-case for knowing whether or not the value was defaulted? You can get the list of defaulted values from the yargs object itself, and at least know whether the value is the same as the default.

**Comment by charlesspitzer** [comment]
i have a command line which needs to enforce: if --arg1 is given, then either --arg2 OR --arg3 is required. if --arg1 has a default, it is instantiated, but there seems to be no way to see, when checking when --arg2 or --arg3 is given, whether --arg1 was instantiated via the default option or whether it was present on the command line.

**Comment by bcoe** [comment]
see: https://github.com/yargs/yargs-parser/issues/203

**Comment by bcoe** [comment]
@charlesspitzer mind taking a look at https://github.com/yargs/yargs/pull/1472, and letting me know if this would do the trick for you?

**Comment by giacomocerquone** [comment]
@bcoe yes, that pr would be ok to me too.
My use case is to warn the user that the tool is defaulting to some value.

**Comment by Lalitha-Iyer** [comment]
I have a slightly different usecase, I would like to filter the user-supplied arguments only and exclude the ones that are defaulting.  I want to then use something like `nconf` to create a override hierarchy, with user arguments at priority 1, followed by file config. 

I will try implement the filter with the changes in PR #1472,

**Comment by bcoe** [comment]
@giacomocerquone unfortunately I've had next to no time for open-source these days, but I'll make an effort to address @coreyfarrell's concerns on that PR "soon".

**Comment by giacomocerquone** [comment]
@bcoe absolutely, take your time, no worries here, we're all grateful for the work and the effort you put in this project :)
